src/components/screens/HistoryScreen/mappers/soroban.tsx

```diff
@@ -0,0 +1,588 @@
+    return processSorobanMint({
+      operation,
+      sorobanAttributes,
+      accountBalances,
+      publicKey,
+      networkDetails,
+      network,
+      stellarExpertUrl,
+      fee,
+      themeColors,
+      baseHistoryItemData,
+    });
+  }
+
+  // Handle token transfer operation
+  if (sorobanAttributes.fnName === SorobanTokenInterface.transfer) {
+    return processSorobanTransfer({
+      operation,
+      sorobanAttributes,
+      publicKey,
+      network,
+      stellarExpertUrl,
+      fee,
+      themeColors,
+      baseHistoryItemData,
+      operationString,
+    });
+  }
+
+  // Default case for other Soroban operations
+  const transactionDetails: TransactionDetails = {
+    operation,
+    transactionTitle: t("Transaction"),
+    transactionType: TransactionType.CONTRACT,
+    status: TransactionStatus.SUCCESS,
+    fee,
+    IconComponent: baseHistoryItemData.IconComponent,
+    ActionIconComponent: baseHistoryItemData.ActionIconComponent,
+    externalUrl: `${stellarExpertUrl}/op/${id}`,
+  };
+
+  return {
+    ...baseHistoryItemData,
+    rowText: operationString,
+    transactionDetails,
+  } as HistoryItemData;
+};
+
+/**
+ * Renders Soroban transfer transaction details
+ */
+export const SorobanTransferTransactionDetailsContent: React.FC<{
+  transactionDetails: TransactionDetails;
+}> = ({ transactionDetails }) => {
+  const { themeColors } = useColors();
+  const tokenAmount = formatTokenAmount(
+    new BigNumber(
+      transactionDetails.contractDetails?.transferDetails?.amount ?? "",
+    ),
+    transactionDetails.contractDetails?.contractDecimals ?? 0,
+  );
+
+  return (
+    <TransactionDetailsContent>
+      <View className="flex-row justify-between">
+        <View>
+          <Text xl primary medium numberOfLines={1}>
+            {tokenAmount} {transactionDetails.contractDetails?.contractSymbol}
+          </Text>
+          <Text md secondary numberOfLines={1}>
+            {/* TODO: add rate */}-
+          </Text>
+        </View>
+        <AssetIcon
+          token={{
+            type: AssetTypeWithCustomToken.CUSTOM_TOKEN,
+            code: transactionDetails.contractDetails?.contractSymbol ?? "",
+            issuer: {
+              key: "",
+            },
+          }}
+          size="lg"
+        />
+      </View>
+
+      <Icon.ChevronDownDouble
+        size={20}
+        color={themeColors.foreground.primary}
+        circle
+        circleBackground={themeColors.background.tertiary}
+      />
+
+      <View className="flex-row justify-between">
+        <View>
+          <Text xl primary medium numberOfLines={1}>
+            {truncatePublicKey({
+              publicKey:
+                transactionDetails.contractDetails?.transferDetails?.to ?? "",
+              length: 4,
+            })}
+          </Text>
+          <Text md secondary numberOfLines={1}>
+            {t("history.transactionDetails.firstTimeSend")}
```

Comment:
we should remove this while we don't have the "firstTimeSend" logic in place. I think it's fine to remove since it's low priority.